src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private static final float RESIZE_THRESHOLD = 1.0f;
+    private static final float RESIZE_MULTIPLIER = 1.5f;
+    private static final int INITIAL_POSITION = 0;
+    private static final int INDEX_SHIFT = 1;
+    private static final String ARRAY_INDEX_EXCEPTION_FORMAT =
+            "Index = [%d] is out of range for ArrayList with size = [%d]";
+    private static final String ELEMENT_NOT_FOUND_FORMAT =
+            "Element = [%s] can't be found in the ArrayList";
+    private Object[] array;
```

Comment:
```suggestion
    private T[] array;
```
that's what we have type parameter for